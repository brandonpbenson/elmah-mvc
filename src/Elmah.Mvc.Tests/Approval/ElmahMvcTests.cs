using System.Text.RegularExpressions;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Elmah.Mvc.Tests.Approval
{
    [UseReporter(typeof(DiffReporter))]
    public class ElmahMvcTests
    {
        private readonly string ElmahMvcAppUrl = "http://localhost:49800";

        [Test]
        public void should_open_elmah_page(string path)
        {
            // do
            var url = ElmahMvcAppUrl + path;

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_stylesheets()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/stylesheet";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/rss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));            
        }

        [Test]
        public void should_open_digest_rss()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/digestrss";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_details()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_not_found_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/detail";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_json_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/json?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        [Test]
        public void should_open_xml_page()
        {
            // do
            var url = ElmahMvcAppUrl + "/admin/elmah/xml?id=d6c13c5b-f57f-41e0-b23d-6de2dfa5bba7";

            // verify
            ApprovalTests.Approvals.VerifyHtml(GetContent(url));
        }

        private string GetContent(string url)
        {
            return HackServertTime(ApprovalTests.Asp.AspApprovals.GetUrlContents(url));
        }

        private string HackServertTime(string content)
        {
            return new Regex("Server time is \\d+:\\d+:\\d+").Replace(content, string.Empty);
        }
    }
}
