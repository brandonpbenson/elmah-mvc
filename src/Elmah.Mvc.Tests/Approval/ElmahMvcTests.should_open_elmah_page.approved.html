<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
	<head><base href="http://localhost:49800/admin">
		<meta http-equiv="X-UA-Compatible" content="IE=EmulateIE7" />
		<title>
			Error log for / on SEEKEY-NOTE (Page #1)
		</title>
		<link rel="stylesheet" type="text/css" href="/admin/elmah/stylesheet" />
		<link rel="alternate" type="application/rss+xml" title="RSS" href="/admin/elmah/rss" />
		<meta http-equiv="refresh" content="60" />

	</head>
	<body>
		<h1 id="PageTitle">
			Error Log for <span id="ApplicationName" title="/">/ on SEEKEY-NOTE</span>
		</h1>
		<ul id="SpeedList">
			<li><a href="/admin/elmah/rss" title="RSS feed of recent errors">RSS Feed</a></li><li><a href="/admin/elmah/digestrss" title="RSS feed of errors within recent days">RSS Digest</a></li><li><a href="/admin/elmah/download" title="Download the entire log as CSV">Download Log</a></li><li><a href="http://elmah.googlecode.com/" title="Documentation, discussions, issues and more">Help</a></li><li><a href="/admin/elmah/about" title="Information about this version and build">About</a></li>
		</ul><p>Errors 1 to 5 of total 5 (page 1 of 1). Start with <a href="/admin/elmah?page=1&amp;size=10" rel="start">10</a>, <a href="/admin/elmah?page=1&amp;size=15" rel="start">15</a>, <a href="/admin/elmah?page=1&amp;size=20" rel="start">20</a>, <a href="/admin/elmah?page=1&amp;size=25" rel="start">25</a>, <a href="/admin/elmah?page=1&amp;size=30" rel="start">30</a>, <a href="/admin/elmah?page=1&amp;size=50" rel="start">50</a> or <a href="/admin/elmah?page=1&amp;size=100" rel="start">100</a> errors per page.</p>
		<table id="ErrorLog" cellspacing="0">
			<tr>
				<th class="host-col" nowrap="nowrap">Host</th><th class="code-col" nowrap="nowrap">Code</th><th class="type-col" nowrap="nowrap">Type</th><th class="error-col" nowrap="nowrap">Error</th><th class="user-col" nowrap="nowrap">User</th><th class="date-col" nowrap="nowrap">Date</th><th class="time-col" nowrap="nowrap">Time</th>
			</tr><tr class="even-row">
				<td class="host-col" nowrap="nowrap">SEEKEY-NOTE</td><td class="code-col" nowrap="nowrap"><span title="Not Found">404</span></td><td class="type-col" nowrap="nowrap"><span title="System.Web.HttpException">Http</span></td><td class="error-col"><span>A public action method &#39;st&#39; was not found on controller &#39;Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController&#39;.</span> <a href="/admin/elmah/detail?id=d5b2db50-0d5b-4c1d-a4dc-30324081191a">Details&hellip;</a></td><td class="user-col" nowrap="nowrap">&nbsp;</td><td class="date-col" nowrap="nowrap"><span title="8 С‡РµСЂРІРЅСЏ 2012 СЂ.">08.06.2012</span></td><td class="time-col" nowrap="nowrap"><span title="18:19:02">18:19</span></td>
			</tr><tr class="odd-row">
				<td class="host-col" nowrap="nowrap">SEEKEY-NOTE</td><td class="code-col" nowrap="nowrap"><span title="Not Found">404</span></td><td class="type-col" nowrap="nowrap"><span title="System.Web.HttpException">Http</span></td><td class="error-col"><span>A public action method &#39;st&#39; was not found on controller &#39;Elmah.Mvc.Web.Areas.Admin.Controllers.ElmahController&#39;.</span> <a href="/admin/elmah/detail?id=79f1c6bf-e3e4-4c31-b0ab-7d99c95cd83a">Details&hellip;</a></td><td class="user-col" nowrap="nowrap">&nbsp;</td><td class="date-col" nowrap="nowrap"><span title="8 С‡РµСЂРІРЅСЏ 2012 СЂ.">08.06.2012</span></td><td class="time-col" nowrap="nowrap"><span title="18:19:02">18:19</span></td>
			</tr><tr class="even-row">
				<td class="host-col" nowrap="nowrap">SEEKEY-NOTE</td><td class="code-col" nowrap="nowrap"><span title="Not Found">404</span></td><td class="type-col" nowrap="nowrap"><span title="System.Web.HttpException">Http</span></td><td class="error-col"><span>The controller for path &#39;/not-found&#39; was not found or does not implement IController.</span> <a href="/admin/elmah/detail?id=24fa4f44-aed0-4c46-920e-e946f623d3ff">Details&hellip;</a></td><td class="user-col" nowrap="nowrap">&nbsp;</td><td class="date-col" nowrap="nowrap"><span title="8 С‡РµСЂРІРЅСЏ 2012 СЂ.">08.06.2012</span></td><td class="time-col" nowrap="nowrap"><span title="17:34:15">17:34</span></td>
			</tr><tr class="odd-row">
				<td class="host-col" nowrap="nowrap">SEEKEY-NOTE</td><td class="code-col" nowrap="nowrap"><span title="Not Found">404</span></td><td class="type-col" nowrap="nowrap"><span title="System.Web.HttpException">Http</span></td><td class="error-col"><span>The controller for path &#39;/not-found&#39; was not found or does not implement IController.</span> <a href="/admin/elmah/detail?id=67b4e325-9342-4e71-8f83-9d700cb17557">Details&hellip;</a></td><td class="user-col" nowrap="nowrap">&nbsp;</td><td class="date-col" nowrap="nowrap"><span title="8 С‡РµСЂРІРЅСЏ 2012 СЂ.">08.06.2012</span></td><td class="time-col" nowrap="nowrap"><span title="17:34:14">17:34</span></td>
			</tr><tr class="even-row">
				<td class="host-col" nowrap="nowrap">SEEKEY-NOTE</td><td class="code-col" nowrap="nowrap">0</td><td class="type-col" nowrap="nowrap"><span title="System.NotImplementedException">NotImplemented</span></td><td class="error-col"><span>The method or operation is not implemented.</span> <a href="/admin/elmah/detail?id=53e3c86e-85f8-4b00-b98c-2354808b15b8">Details&hellip;</a></td><td class="user-col" nowrap="nowrap">&nbsp;</td><td class="date-col" nowrap="nowrap"><span title="8 С‡РµСЂРІРЅСЏ 2012 СЂ.">08.06.2012</span></td><td class="time-col" nowrap="nowrap"><span title="17:34:05">17:34</span></td>
			</tr>
		</table><p></p>
		<p id="Footer"><span>Powered by <a href="http://elmah.googlecode.com/">ELMAH</a>, version 1.2.14706.955. Copyright (c) 2004, Atif Aziz. All rights reserved. Licensed under <a href="http://www.apache.org/licenses/LICENSE-2.0">Apache License, Version 2.0</a>. </span>Server date is Friday, 08 June 2012. . All dates and times displayed are in the FLE Daylight Time zone. This log is provided by the XML File-Based Error Log.</p>
	</body>

</html>
